amgut/lib/data_access/tests/test_ag_data_access.py

```diff
@@ -0,0 +1,162 @@
+
+    def test_create_AGDataAccess(self):
+        raise NotImplementedError()
+
+    def test_delete_AGDataAccess(self):
+        raise NotImplementedError()
+
+    def test_get_cursor(self):
+        raise NotImplementedError()
+
+    def test_open_connection(self):
+        raise NotImplementedError()
+
+    def test_get_col_names_from_cursor(self):
+        raise NotImplementedError()
+
+    def test_authenticateWebAppUser(self):
+        raise NotImplementedError()
+
+    def test_addAGLogin(self):
+        raise NotImplementedError()
+
+    def test_getAGBarcodeDetails(self):
+        raise NotImplementedError()
+
+    def test_getAGKitDetails(self):
+        raise NotImplementedError()
+
+    def test_registerHandoutKit(self):
+        raise NotImplementedError()
+
+    def test_deleteAGParticipantSurvey(self):
+        raise NotImplementedError()
+
+    def test_getConsent(self):
+        raise NotImplementedError()
+
+    def test_logParticipantSample(self):
+        raise NotImplementedError()
+
+    def test_deleteSample(self):
+        raise NotImplementedError()
+
+    def test_getHumanParticipants(self):
+        raise NotImplementedError()
+
+    def test_is_old_survey(self):
+        raise NotImplementedError()
+
+    def test_updateVioscreenStatus(self):
+        raise NotImplementedError()
+
+    def test_getAnimalParticipants(self):
+        raise NotImplementedError()
+
+    def test_getParticipantSamples(self):
+        raise NotImplementedError()
+
+    def test_getEnvironmentalSamples(self):
+        raise NotImplementedError()
+
+    def test_getAvailableBarcodes(self):
+        raise NotImplementedError()
+
+    def test_verifyKit(self):
+        raise NotImplementedError()
+
+    def test_getMapMarkers(self):
+        raise NotImplementedError()
+
+    def test_handoutCheck(self):
+        raise NotImplementedError()
+
+    def test_check_access(self):
+        raise NotImplementedError()
+
+    def test_getAGKitIDsByEmail(self):
+        raise NotImplementedError()
+
+    def test_ag_set_pass_change_code(self):
+        raise NotImplementedError()
+
+    def test_ag_update_kit_password(self):
+        skid = 'test'
+        # Check and set pass_reset_code if needed
+        sql = '''UPDATE ag.ag_kit
+                 SET pass_reset_code = %s
+                 WHERE supplied_kit_id = %s AND pass_reset_code IS NULL
+                 RETURNING pass_reset_code'''
+        re_code = self.conn_handler.execute_fetchone(sql, ['T3ST', skid])[0]
+        # Verify password is not what we are changing it to
+        sql = 'SELECT kit_password FROM ag.ag_kit WHERE supplied_kit_id = %s'
+        old_pass = self.conn_handler.execute_fetchone(sql, [skid])[0]
+        self.assertFalse(bcrypt.verify('password', old_pass))
+
+        # Verify password change
+        test_sql = '''SELECT kit_password, pass_reset_code
+                      FROM ag.ag_kit
+                      WHERE supplied_kit_id = %s'''
+        try:
```

Comment:
As it says, you need to reset the database to what it was no matter if the tests fail or not, hence the try/finally